### PR TITLE
tests integ: Fix nm VLAN tests by refreshing nm client

### PR DESCRIPTION
**Pull request by EdDev**
Remove the sleep introduced in previous patched and add an NM client
refresh.
While outside the mainloop, the client needs to be re-created in order
to retrieve updated information form NM.

### CODE REVIEW

File: tests/integration/nm/vlan_test.py [hunk 1]
```diff
@@ -50,6 +46,7 @@ def _vlan_interface(state):
 
 
 def _get_vlan_current_state(vlan_desired_state):
+    nm.nmclient.client(refresh=True)
```
Comment (tyll) [inline tests/integration/nm/vlan_test.py]: IMHO it would be better to do this in the get_device_by_name() function but since it segfaults there, this is an acceptable workaround. Please add a FIXME, though, so we remove this eventually.

Comment (EdDev) [reply tests/integration/nm/vlan_test.py]: It cannot be done in the production NM package code because there we are already operating inside the mainloop (for apply at least). I guess that is the cause of the segfaults.

Regarding the FIXME, I do not think we can "fix" it. In the NM integration tests we run the low level NM api, and create the mainloop ourself in the test. This last assert checks if the interface has been removed (those action have been executed in the mainloop), and it does not need a mainloop, just a client refresh.
Bottom line, there is no fix here unless you can think of something else we can do to make this nicer.

**Comment by EdDev** [comment]
Now it fails on something else :)

**Comment by EdDev** [comment]
ci test please

File: tests/integration/nm/vlan_test.py (hunk 1 above)
Comment (tyll) [reply tests/integration/nm/vlan_test.py]: Keeping the mainloop running would ensure that the client will be refreshed AFAIU, so it would be a fix... But I see that it would be difficult to do this in the test code as well.
